File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
class Rectangle:
        def __init__(self, width=0, height=0):
                self.width = width
                self.height = height
        def width(self, value):
                if type(self.width) is not int:
                        raise TypeError
                elif self.width < 0:
                        raise ValueError
                else:
                        return self.width
        def height(self, value):
                if type(self.height) is not int:
                        raise TypeError
                elif self.height < 0:
                        raise ValueError
        def area(self,value):
                self.area = self.width * self.height
                return self.area
        def perimeter(self, value):
                self.perimeter = 2*(self.height + self.width)
		if (width = 0) or (height = 0):
			self.perimeter = 0
                return self.perimeter

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
/* A class Rectangle that defines a rectangle. */
module.exports = class Rectangle {
  /* constructor takes in 2 arguments w and h */
  constructor (w, h) {
    /* checks if width and height values are greater than equal to 0 */
    if (w > 0 && h > 0) {
    /* assigns the given value to its respective variable */
      [this.width, this.height] = [w, h];
    }
  }

  print () {
    /* loop thru the height of the rectangle */
    for (let i = 0; i < this.height; i++) {
      /* print out 'X' repeating it the number of it's width times */
      console.log('X'.repeat(this.width));
    }
  }
};
